packages/addon-shim/src/index.ts

```diff
@@ -158,8 +158,12 @@ export function addonV1Shim(directory: string, options: ShimOptions = {}) {
       return isInside(directory, appInstance.project.root);
     },
 
-    registerV2Addon(name: string, root: string): void {
-      autoImportInstance!.registerV2Addon(name, root);
+    registerV2Addon(this: AddonInstance, name: string, root: string): void {
+      if (autoImportInstance) {
+        autoImportInstance.registerV2Addon(name, root);
+      } else {
+        (this.parent as EAI2Instance).registerV2Addon(name, root);
```

Comment:
How do we know it's safe to call this on the parent?

Reply:
That's the biggest question mark I have, quote from the issue above:

> Why I'm saying this is only a suggested fix: I'm not sure, if this piece of code is in the intention of the comment or if that means a different implementation (given I have basically no real knowledge about embroider).

which basically targets this line.

But following the analysis I put in the issue, there is either a `autoImportInstance` or `this.parent` with `registerV2Addon` available.

Reply:
Potentially, you can safeguard this. But I'd say, it would be better to throw an error over silently do nothing and not registering the addon at all, without a dev knowing about it ?!?

Reply:
It's safe once refactored inside the `if (this.parent.pkg['ember-addon']?.version ?? 1) < 2)` check.